### PR TITLE
Add support for custom ports to bind haproxy

### PR DESCRIPTION
**Pull request by m-yosefpor**
fixes: https://github.com/openshift/cluster-ingress-operator/issues/633

related issue: https://github.com/openshift/api/issues/964

Checklist:
- [x] controller
- [x] unit test
- [ ] e2e test
- [ ] https://github.com/openshift/api/pull/965 is merged.

### CODE REVIEW

File: pkg/operator/controller/ingress/deployment.go [hunk 1]
```diff
@@ -70,6 +70,11 @@ const (
 	RouterHAProxyThreadsEnvName      = "ROUTER_THREADS"
 	RouterHAProxyThreadsDefaultValue = 4
 
+	RouterHAProxyServiceHTTPPortEnvName       = "ROUTER_SERVICE_HTTP_PORT"
+	RouterHAProxyServiceHTTPPortDefaultValue  = 80
+	RouterHAProxyServiceHTTPsPortEnvName      = "ROUTER_SERVICE_HTTP_PORT"
```
Comment (frobware) [inline pkg/operator/controller/ingress/deployment.go]: This should be `ROUTER_SERVICE_**HTTPS**_PORT`.

**Comment by m-yosefpor** [comment]
@frobware Thanks for your review. I'm waiting to hear from @pweil-  in https://github.com/openshift/api/pull/965 to see what will product team of Openshift decides about this feature. This PR would be a WIP until then.

I also need to change a couple of more things to provide SNI_PORT and NO_SNI_PORT options and validation (to avoid conflicts) + writing e2e tests for this feature.

**Comment by m-yosefpor** [comment]
The openshift team decided to include this feature in a future release. So I'm going to start working on this again.

**Comment by pweil-** [comment]
Red Hat sponsored projects and open source communities are subject to and must comply with the U.S. Export Regulations. These regulations prohibit Red Hat software, services, and support from being exported, re-exported or transferred, either directly or indirectly, to a prohibited destination and/or country under U.S. sanctions administered by the U.S. Department of Treasury, Office of Foreign Assets Control ("OFAC"). Because there is indication that your contribution is from an embargoed country and would violate the U.S. Export Regulations to accept, your access has been restricted. You may review OFAC’s Sanctions Programs and Information at: https://www.treasury.gov/resource-center/sanctions/pages/default.aspx and Red Hat’s export compliance policies and guidelines at: https://www.redhat.com/en/about/export-control-product-matrix.

**Pull request closed without merging**